packages/alfa-rules/test/sia-r11/rule.spec.tsx

```diff
@@ -16,32 +16,46 @@ test(`evaluate() passes a link with a name given by content`, async (t) => {
   ]);
 });
 
-test(`evaluate() fails a link with no name`, async (t) => {
-  const target = <a href="#"></a>;
+test(`evaluate() passes an image link with name given by the alt text`, async (t) => {
+  const target = (
+    <a href="#">
+      <img src="foo.jpg" alt="Foo" />
+    </a>
+  );
 
   const document = h.document([target]);
 
   t.deepEqual(await evaluate(R11, { document }), [
-    failed(R11, target, { 1: Outcomes.HasNoName }),
+    passed(R11, target, { 1: Outcomes.HasName }),
   ]);
 });
 
-test(`evaluate() is inapplicable when there is no link`, async (t) => {
-  const document = h.document([<div></div>]);
+test(`evaluate() fails a link with no name`, async (t) => {
+  const target = <a href="#"></a>;
 
-  t.deepEqual(await evaluate(R11, { document }), [inapplicable(R11)]);
+  const document = h.document([target]);
+
+  t.deepEqual(await evaluate(R11, { document }), [
+    failed(R11, target, { 1: Outcomes.HasNoName }),
+  ]);
 });
 
-test(`evaluate() passes an image link with name given by the alt text`, async (t) => {
+test(`evaluate() fails an 'a' element, which inherits semantically a link role, with no accessible name`, async (t) => {
   const target = (
-    <a href="#">
-      <img src="foo.jpg" alt="Foo" />
+    <a href="#" role="I am giving a link role">
+      <img src="Foo.jpg" alt="" />
     </a>
```

Comment:
This should be an existing role 😅 
Also would be better with a non`<a>` element. Currently, the role is invalid, so it defaults to the implicit role of `link`, and this does not test the change.
```suggestion
    <span role="doc-biblioref">
      <img src="Foo.jpg" alt="" />
    </span>
```